src/App.tsx

```diff
@@ -1,27 +1,46 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.scss';
 import { GoodsList } from './GoodsList';
+import { Good } from './types/Good';
+import { getAll, get5First, getRedGoods } from './api/goods';
 
-// import { getAll, get5First, getRed } from './api/goods';
-// or
-// import * as goodsAPI from './api/goods';
+export const App: React.FC = () => {
+  const [goods, setGoods] = useState<Good[]>([]);
 
-export const App: React.FC = () => (
-  <div className="App">
-    <h1>Dynamic list of Goods</h1>
+  const loadGoods = (goodsFromServer: () => Promise<Good[]>) => {
+    goodsFromServer()
+      .then((data: Good[]) => setGoods(data));
```

Comment:
you should handle error
Also, would be better to use `try / catch` constructions
![image](https://github.com/mate-academy/react_dynamic-list-of-goods/assets/91411692/f3ec4678-cef1-4978-9ad4-08f09b941520)